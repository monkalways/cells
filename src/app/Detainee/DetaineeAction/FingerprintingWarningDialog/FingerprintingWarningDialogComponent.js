import React from 'react';
import PropTypes from 'prop-types';
import {
  Button, Dialog, DialogActions, DialogTitle,
} from '@material-ui/core';

const propTypes = {
  isDialogOpen: PropTypes.bool.isRequired,
  handleCancel: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired,
  handleIgnore: PropTypes.func.isRequired,
};

const FingerprintingWarningDialogComponent = ({
  isDialogOpen,
  handleCancel,
  handleClose,
  handleIgnore,
}) => (
  <Dialog open={isDialogOpen} onClose={handleClose}>
    <React.Fragment>
      <DialogTitle>
        {'Warning: Fingerprinting has not been completed.'}
      </DialogTitle>

      <DialogActions>
        <Button id="cancelButton" onClick={handleCancel} color="primary">
          Cancel
        </Button>
        <Button
          id="ignoreButton"
          onClick={handleIgnore}
          color="primary"
          variant="contained"
          autoFocus
        >
          Ignore
        </Button>
      </DialogActions>
    </React.Fragment>
  </Dialog>
);
FingerprintingWarningDialogComponent.propTypes = propTypes;

export default FingerprintingWarningDialogComponent;
