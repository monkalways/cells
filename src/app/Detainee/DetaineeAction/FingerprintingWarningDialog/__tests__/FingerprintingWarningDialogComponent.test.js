import React from 'react';
import { shallow } from 'enzyme';
import { Dialog } from '@material-ui/core';

import FingerprintingWarningDialogComponent from '../FingerprintingWarningDialogComponent';

describe('FingerprintingWarningDialogComponent', () => {
  let isDialogOpen;
  let handleCancel;
  let handleClose;
  let handleIgnore;

  beforeEach(() => {
    isDialogOpen = true;
    handleCancel = jest.fn();
    handleClose = jest.fn();
    handleIgnore = jest.fn();
  });

  const setup = () => shallow(<FingerprintingWarningDialogComponent
    isDialogOpen={isDialogOpen}
    handleCancel={handleCancel}
    handleClose={handleClose}
    handleIgnore={handleIgnore}
  />);

  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should handleClose', () => {
    const wrapper = setup();
    wrapper.find(Dialog).simulate('close');
    expect(handleClose).toBeCalled();
  });

  it('should handleCancel', () => {
    const wrapper = setup();
    wrapper.find('#cancelButton').simulate('click');
    expect(handleCancel).toBeCalled();
  });

  it('should handleIgnore', () => {
    const wrapper = setup();
    wrapper.find('#ignoreButton').simulate('click');
    expect(handleIgnore).toBeCalled();
  });
});
