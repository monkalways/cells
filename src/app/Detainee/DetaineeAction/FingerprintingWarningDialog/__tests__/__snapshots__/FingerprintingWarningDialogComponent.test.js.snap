// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FingerprintingWarningDialogComponent should render correctly 1`] = `
<WithStyles(Dialog)
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(DialogTitle)>
    Warning: Fingerprinting has not been completed.
  </WithStyles(DialogTitle)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      id="cancelButton"
      onClick={[MockFunction]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      autoFocus={true}
      color="primary"
      id="ignoreButton"
      onClick={[MockFunction]}
      variant="contained"
    >
      Ignore
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
